goss-testing/scripts/check_interface_mac_matches_bond.sh

```diff
@@ -61,67 +51,49 @@ Exits 0 if this is the case. Exits non-0 if not, or if there are any errors.
     error "$@"
 }
 
-# cmdfail <script exit code> <command exit code> <command + args>
-function cmdfail
-{
-    error $1 "Command failed (exit code $2): $3"
-}
+INTERFACES=( "$@" )
 
-if [[ $# -eq 0 ]]; then
+if [[ ${#INTERFACES[@]} -eq 0 ]]; then
     usage 5 "No interface specified"
-elif [[ $# -gt 1 ]]; then
-    usage 10 "Too many arguments"
-elif [[ -z $1 ]]; then
-    usage 15 "Interface argument may not be null"
+elif [[ -z ${#INTERFACES[@]} ]]; then
+    usage 15 "Interface argument(s) may not be null"
 fi
 
-TMPFILE=/tmp/check_interface_mac_matches_bound.$$.$RANDOM.tmp
-INTERFACE="$1"
-
-echo "Checking that MAC address of interface '$INTERFACE' matches MAC address of bond0..."
-echo
-
-# First print the entire output of the 'ip addr show' command, mainly for potential use in debugging
-# test failures.
-
-echo "# ip addr show"
-ip addr show ||
-    cmdfail 25 $? "ip addr show"
-
-echo
-
-# Get the MAC address of the interface. 
-echo "# ip addr show dev '$INTERFACE'"
-ip addr show dev "$INTERFACE" | tee "$TMPFILE" || 
-    cmdfail 30 $? "ip addr show dev '$INTERFACE' | tee '$TMPFILE'"
-
-INT_MAC=$( grep link/ether "$TMPFILE" | awk '{ print $2 }' ) || 
-    cmdfail 35 $? "grep link/ether '$TMPFILE' | awk '{ print \$2 }'"
-echo "Interface ($INTERFACE) MAC address = $INT_MAC"
-
-echo
-
 # Get the MAC address of bond0.
-echo "# ip addr show dev bond0"
-ip addr show dev bond0 | tee "$TMPFILE" || 
-    cmdfail 50  $? "ip addr show dev bond0 | tee '$TMPFILE'"
-BND_MAC=$( grep link/ether "$TMPFILE" | awk '{ print $2 }' ) || 
-    cmdfail 55 $? "grep link/ether '$TMPFILE' | awk '{ print \$2 }'"
-echo "bond0 MAC address = $BND_MAC"
+BND_MAC=""
+BND_MAC="$( ip addr show dev bond0 | grep link/ether | awk '{ print $2 }' )" || 
+    error 55 "bond0 MAC could not be retrieved"
 
+for INTERFACE in "${INTERFACES[@]}"
+do
+echo "Checking that MAC address of interface '$INTERFACE' matches MAC address of bond0..."
 echo
 
-# Validate that we actually got a MAC address for each
-[[ -n $INT_MAC ]] || error 70 "No MAC address found for $INTERFACE"
-[[ -n $BND_MAC ]] || error 75 "No MAC address found for bond0"
-
-# Finally, validate that they match
-[[ $INT_MAC == $BND_MAC ]] || error 90 "MAC address of $INTERFACE does not match MAC address of bond0"
+# if the interface does not show, it might be the option CAN
+if ! eval ip addr show dev "$INTERFACE" 1>/dev/null; then
```

Comment:
I don't think you need the "eval" here, but equally I don't think it hurts anything.